### PR TITLE
feat: add new condition operators for Choice node

### PR DESCRIPTION
**Pull request by TrachukT**
> [!IMPORTANT]
> Add new string condition operators to Choice node and update tests for comprehensive coverage.
> 
>   - **Behavior**:
>     - Adds new string condition operators: `STRING_STARTS_WITH`, `STRING_ENDS_WITH`, `STRING_CONTAINS`, `STRING_REGEXP` to `ConditionOperator` in `types.py`.
>     - Updates `evaluate()` in `operators.py` to handle new operators, including error handling for invalid regex.
>   - **Tests**:
>     - Adds tests for new operators in `test_choice.py`, including edge cases and negation scenarios.
>     - Tests cover `STRING_STARTS_WITH`, `STRING_ENDS_WITH`, `STRING_CONTAINS`, `STRING_REGEXP` with various input data.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=dynamiq-ai%2Fdynamiq&utm_source=github&utm_medium=referral)<sup> for 2444d67771928ec83a35bc34d669c5ec21065d90. You can [customize](https://app.ellipsis.dev/dynamiq-ai/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: dynamiq/nodes/operators/operators.py [hunk 1]
```diff
@@ -146,6 +147,15 @@ def evaluate(cond: ChoiceCondition, input_data: Any) -> bool:
             return (value < cond.value) == (not cond.is_not)
         elif cond.operator == ConditionOperator.STRING_LESS_THAN_OR_EQUALS:
             return (value <= cond.value) == (not cond.is_not)
+        elif cond.operator == ConditionOperator.STRING_STARTS_WITH:
```
Comment (acoola) [inline dynamiq/nodes/operators/operators.py:150]: Let's add `STRING_ENDS_WITH` too
